### DISCUSSION TITLE
ProfileScreen should ask for confirmation before deleting the account and unlinking a "sign in method"

### DISCUSSION THREAD

**Discussion by guplem** [firebase/flutterfire#9679] [Feature request]
In the default profile screen the user can unlink the sign in methods they desire.

However, no confirmation is asked before unlinking. A miss click might unlink a user from a service method.
![image](https://user-images.githubusercontent.com/11029629/194037081-e06ca5d0-5ec5-4858-8d22-8a000b02f8e0.png)

Making things worse, you can remove all the sign in methods and not be able to log in again:
![image](https://user-images.githubusercontent.com/11029629/194037426-5995ebd6-db95-490c-80cc-c490b064e7da.png)

Even if the user tries to re-enable it, they are asked to log in again to re-enable a sign in method:
![image](https://user-images.githubusercontent.com/11029629/194037533-dc70111b-d169-4056-be84-6dbaa71f9288.png)

And I just realized you can even delete the account without asking for confirmation as well! (if you don't have any sign in method)

**Comment by graemep-nz** [comment]
It's related to this, maybe some duplication
https://github.com/firebase/flutterfire/issues/9229

**Comment by amoschoomy** [comment]
>I just realized you can even delete the account without asking for confirmation as well! (if you don't have any sign in method)

Seconded on this, this makes it very dangerous to include the profile screen in production apps. Hopefully this gets the attention it needed.

**Reply by guplem** [reply]
I ended up creating a [dedicated issue](https://github.com/firebase/flutterfire/issues/11074) since no attention from firebase seemed to be drowned in this discussion

**Comment by EnduringBeta** [comment]
This issue makes me uncomfortable using `ProfileScreen`. It would be really beneficial to have a dialog that ensures the user knows what they're doing when they tap an innocuous icon.

Edit: And also, this could just be my ignorance, but my current implementation only has email/password sign-in. If they tap the icon, not knowing what they've done, and then at some point sign out, I believe they've abandoned their account and its data? That's really bad to me.

**Reply by EnduringBeta** [reply]
I've moved to implementing my own profile route, by the way.
